WalletWasabi/Services/UpdateManager.cs

```diff
@@ -1,329 +1,252 @@
+			await ProcessReleaseEventsAsync(cancellationToken).ConfigureAwait(false);
 		}
-		catch (Exception ex)
+		finally
 		{
-			Logger.LogError("Getting new update failed with error.", ex);
+			// Ensure we disconnect regardless of the outcome
+			await _nostrClient.DisconnectAsync(cancellationToken).ConfigureAwait(false);
 		}
 	}
 
-	/// <summary>
-	/// Get or download installer for the newest release.
-	/// </summary>
-	private async Task<(string filePath, Version newVersion)> GetInstallerAsync(ReleaseInfo info, CancellationToken cancellationToken)
+	private async Task ProcessReleaseEventsAsync(CancellationToken cancellationToken)
 	{
-		var sha256SumsFilePath = Path.Combine(_installerDir, "SHA256SUMS.asc");
-
-		// This will throw InvalidOperationException in case of invalid signature.
-		await DownloadAndValidateWasabiSignatureAsync(sha256SumsFilePath, info.AssetDownloadLinks, cancellationToken).ConfigureAwait(false);
-
-		var installerFilePath = Path.Combine(_installerDir, info.InstallerFileName);
+		using var sixtySeconds = new CancellationTokenSource(TimeSpan.FromSeconds(60));
+		using var linkedCancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, sixtySeconds.Token);
 
 		try
 		{
-			if (!File.Exists(installerFilePath))
+			// Read all the events as an array
+			var releases = await _nostrClient.EventsReader
+				.ReadAllAsync(linkedCancellationTokenSource.Token)
+				.ToArrayAsync(linkedCancellationTokenSource.Token)
+				.ConfigureAwait(false);
+
+			// Find release with version greater than current version
+			var latestRelease = releases
+				.Where(x => x.Version > Constants.ClientVersion)
+				.MaxBy(x => x.Version);
+
+			if(latestRelease is not null)
 			{
-				EnsureToRemoveCorruptedFiles();
+				Logger.LogInfo($"New version found: {latestRelease.Version}");
 
-				Logger.LogInfo($"Trying to download new version: {info.LatestClientVersion}");
+				// Notify about new version via event bus
+				var updateStatus = new UpdateStatus(ClientVersion: latestRelease.Version, ClientUpToDate: false, IsReadyToInstall: false);
+				eventBus.Publish(new NewSoftwareVersionAvailable(updateStatus));
 
-				// Get file stream and copy it to downloads folder to access.
-				using HttpRequestMessage request = new(HttpMethod.Get, info.InstallerDownloadUrl);
-				using HttpResponseMessage response = await _githubHttpClient.SendAsync(request, cancellationToken).ConfigureAwait(false);
-				byte[] installerFileBytes = await response.Content.ReadAsByteArrayAsync(cancellationToken).ConfigureAwait(false);
-
-				Logger.LogInfo("Installer downloaded, copying...");
-
-				using MemoryStream stream = new(installerFileBytes);
-				await CopyStreamContentToFileAsync(stream, installerFilePath, cancellationToken).ConfigureAwait(false);
+				// Download the new version
+				await releaseDownloader(latestRelease, cancellationToken).ConfigureAwait(false);
```

Comment:
Those calls should be reversed I believe. First download and verify the release, then update the status

Reply:
I prefer to notify that there is a new version and then notify again using a different event once the installer was finally downloaded and verified.